src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingMessagingService.java

```diff
@@ -188,13 +193,16 @@ public void notifyUserAboutNewResult(Result result, ProgrammingExerciseParticipa
         if (participation instanceof ProgrammingExerciseStudentParticipation studentParticipation) {
             // do not try to report results for template or solution participations
             ltiNewResultService.ifPresent(newResultService -> newResultService.onNewResult(studentParticipation));
-            // Inform Iris about the submission status
+            // Inform Iris about the submission status (when certain conditions are met)
             notifyIrisAboutSubmissionStatus(result, studentParticipation);
         }
     }
 
     /**
      * Notify Iris about the submission status for the given result and student participation.
+     * Only if the user has accepted Iris, the exercise is not an exam exercise, and the exercise chat is enabled in the exercise settings
```

Comment:
Really a minor, but it took me a few reads to understand, that "only if" is related to the first sentence. Maybe sth like might be more descriptive:

```suggestion
     * Only notifies if the user has accepted Iris, the exercise is not an exam exercise, and the exercise chat is enabled in the exercise settings
```